src/docfx/lib/moniker/EvaluatorWithMonikersVisitor.cs

```diff
@@ -65,44 +65,45 @@ public IEnumerable<string> Visit(LogicExpression expression)
 
         private static Dictionary<string, MonikerProductInfo> InitializeMonikers(IEnumerable<Moniker> monikers)
         {
-            var monikerName = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
-            var productNameDictionary = new Dictionary<string, List<Moniker>>(StringComparer.OrdinalIgnoreCase);
+            var monikerNameList = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            var productNameDictionary = new Dictionary<string, List<string>>(StringComparer.OrdinalIgnoreCase);
 
             foreach (var moniker in monikers)
             {
-                if (monikerName.Contains(moniker.MonikerName))
+                if (string.IsNullOrEmpty(moniker.Name))
                 {
-                    throw Errors.MonikerNameConflict(moniker.MonikerName).ToException();
+                    throw Errors.InvalidMonikerDefinition("Moniker name cannot be null or empty").ToException();
                 }
-                else
+                if (string.IsNullOrEmpty(moniker.Product))
+                {
+                    throw Errors.InvalidMonikerDefinition("Product name cannot be null or empty").ToException();
+                }
+                if (monikerNameList.Contains(moniker.Name))
```

Comment:
It can be 
```csharp
if(!monikerNameList.Add(moniker.Name))
```
And remove the line below

Reply:
:thumbsup: 

---
In reply to: [230656500](https://github.com/dotnet/docfx/pull/3575#discussion_r230656500) [](ancestors = 230656500)